.github/workflows/checks.yaml

```diff
@@ -18,3 +18,58 @@ jobs:
         run: node ./scripts/check-conventional-commits/command.mjs
         env:
           REF: ${{ github.ref }}
+
+      # for preview deployment for specific project, add vercel project id in environment section
+      - name: Preview Deployment
+        id: deploy
+        run: |
+          yarn global add vercel
+          yarn run deploy
+        env:
+          REF: ${{ github.ref }}
+          GH_TOKEN: ${{ github.token }}
+          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
+          VERCEL_PROJECT_WIDGET_CONFIG: ${{ secrets.VERCEL_PROJECT_WIDGET_CONFIG }} # widget playground
+          VERCEL_PROJECT_WIDGET_APP: ${{ secrets.VERCEL_PROJECT_WIDGET_APP }} # widget app
+          ENABLE_PREVIEW_DEPLOY: true
+
+    outputs:
+      # the structure of output variable is {packageNameWithoutScope}-url like: widget-app-url
+      app_url: ${{ steps.deploy.outputs.widget-app-url }}
+      playground_url: ${{ steps.deploy.outputs.widget-playground-url }}
+
+  # add job for each project that you want has preview deployment
+  app-preview:
+    runs-on: ubuntu-latest
+    needs: check
+    environment:
+      name: app-preview
+      url: ${{ steps.seturl.outputs.url }}
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
Is this step needed?

Reply:
you were right
We didn't need this step,
I deleted it